Lib/os.py

```diff
@@ -1118,7 +1118,7 @@ def _fspath(path):
     fspath.__name__ = "fspath"
 
 
-class PathLike(abc.ABC):
```

Comment:
I don't think that changing the base class of existing types is a good idea for stability. Some people might expect that `PathLike` is a subclass of `ABC` and use this logic somewhere in their code.

Reply:
Whether or not it's correct for developers to be depending on that, I agree that this could break people's code in subtle ways, and that the benefits of changing the base class here are very marginal. This is the standard library, so we need to be especially careful when it comes to backwards compatibility.

I agree with @sobolevn that we should add it to the protocol allowlist, but not change the base class here.

Reply:
Python is a big project which is widely used.
`ABC` -> `ABCMeta` *might* break some code without a strong reason.
On a particular project we could discuss should we respect `issubclass(cls, ABC)` or discourage the check; in Python itself we are avoiding breaking changes what modify public API if it is possible.

Reply:
My question may be out of scope though: it doesn't change the fact that we change the base class.
I still think it's worth sorting out though.

Reply:
Oh, sorry, I caught some delay.

Thanks, I see your point. TIL! I will definitely note such things up in the future then :)

Reply:
That all makes sense; I've backed out that part of the change.